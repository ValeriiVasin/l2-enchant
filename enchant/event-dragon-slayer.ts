import { createEnchantMap } from '../helpers/create-enchant-map';
import { Item } from '../items';
import { EnchantStrategy } from '../types';

// https://l2central.info/classic/events_and_promos/1136.html?lang=ru
const chances = [100, 100, 100, ...Array(12).fill(70)];
const items = [
  Item.EventDragonSlayerSword,
  Item.EventDragonSlayerSword_1,
  Item.EventDragonSlayerSword_2,
  Item.EventDragonSlayerSword_3,
  Item.EventDragonSlayerSword_4,
  Item.EventDragonSlayerSword_5,
  Item.EventDragonSlayerSword_6,
  Item.EventDragonSlayerSword_7,
  Item.EventDragonSlayerSword_8,
  Item.EventDragonSlayerSword_9,
  Item.EventDragonSlayerSword_10,
  Item.EventDragonSlayerSword_11,
  Item.EventDragonSlayerSword_12,
  Item.EventDragonSlayerSword_13,
  Item.EventDragonSlayerSword_14,
  Item.EventDragonSlayerSword_15,
];

export const eventDragonSlayerEnchantMap = createEnchantMap({
  items,
  chances,
  enchantItem: Item.EventEnchantDragonSlayer,
  strategy: EnchantStrategy.Destroy,
});
