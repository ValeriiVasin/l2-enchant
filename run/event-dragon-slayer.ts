import { itemsRange } from '../helpers/items-range';
import { run } from '../helpers/run';
import { Item } from '../items';

run(Item.EventDragonSlayerSword, itemsRange(Item.EventDragonSlayerSword_7, Item.EventDragonSlayerSword_15), 100_000);
