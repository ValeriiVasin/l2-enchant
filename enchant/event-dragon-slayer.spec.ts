import { eventDragonSlayerEnchantMap } from './event-dragon-slayer';

it('enchant map: event-dragon-slayer', () => {
  expect(eventDragonSlayerEnchantMap).toMatchSnapshot();
});
