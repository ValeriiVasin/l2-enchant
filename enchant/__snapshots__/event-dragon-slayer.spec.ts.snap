// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enchant map: event-dragon-slayer 1`] = `
Map {
  "Меч Убийца Драконов" => Object {
    "cost": "Свиток Модификации Убийц Драконов",
    "item": "Меч Убийца Драконов",
    "results": Array [
      Object {
        "chance": 100,
        "item": "Меч Убийца Драконов +1",
      },
      Object {
        "chance": 0,
        "item": "Nothing",
      },
    ],
  },
  "Меч Убийца Драконов +1" => Object {
    "cost": "Свиток Модификации Убийц Драконов",
    "item": "Меч Убийца Драконов +1",
    "results": Array [
      Object {
        "chance": 100,
        "item": "Меч Убийца Драконов +2",
      },
      Object {
        "chance": 0,
        "item": "Nothing",
      },
    ],
  },
  "Меч Убийца Драконов +2" => Object {
    "cost": "Свиток Модификации Убийц Драконов",
    "item": "Меч Убийца Драконов +2",
    "results": Array [
      Object {
        "chance": 100,
        "item": "Меч Убийца Драконов +3",
      },
      Object {
        "chance": 0,
        "item": "Nothing",
      },
    ],
  },
  "Меч Убийца Драконов +3" => Object {
    "cost": "Свиток Модификации Убийц Драконов",
    "item": "Меч Убийца Драконов +3",
    "results": Array [
      Object {
        "chance": 70,
        "item": "Меч Убийца Драконов +4",
      },
      Object {
        "chance": 30,
        "item": "Nothing",
      },
    ],
  },
  "Меч Убийца Драконов +4" => Object {
    "cost": "Свиток Модификации Убийц Драконов",
    "item": "Меч Убийца Драконов +4",
    "results": Array [
      Object {
        "chance": 70,
        "item": "Меч Убийца Драконов +5",
      },
      Object {
        "chance": 30,
        "item": "Nothing",
      },
    ],
  },
  "Меч Убийца Драконов +5" => Object {
    "cost": "Свиток Модификации Убийц Драконов",
    "item": "Меч Убийца Драконов +5",
    "results": Array [
      Object {
        "chance": 70,
        "item": "Меч Убийца Драконов +6",
      },
      Object {
        "chance": 30,
        "item": "Nothing",
      },
    ],
  },
  "Меч Убийца Драконов +6" => Object {
    "cost": "Свиток Модификации Убийц Драконов",
    "item": "Меч Убийца Драконов +6",
    "results": Array [
      Object {
        "chance": 70,
        "item": "Меч Убийца Драконов +7",
      },
      Object {
        "chance": 30,
        "item": "Nothing",
      },
    ],
  },
  "Меч Убийца Драконов +7" => Object {
    "cost": "Свиток Модификации Убийц Драконов",
    "item": "Меч Убийца Драконов +7",
    "results": Array [
      Object {
        "chance": 70,
        "item": "Меч Убийца Драконов +8",
      },
      Object {
        "chance": 30,
        "item": "Nothing",
      },
    ],
  },
  "Меч Убийца Драконов +8" => Object {
    "cost": "Свиток Модификации Убийц Драконов",
    "item": "Меч Убийца Драконов +8",
    "results": Array [
      Object {
        "chance": 70,
        "item": "Меч Убийца Драконов +9",
      },
      Object {
        "chance": 30,
        "item": "Nothing",
      },
    ],
  },
  "Меч Убийца Драконов +9" => Object {
    "cost": "Свиток Модификации Убийц Драконов",
    "item": "Меч Убийца Драконов +9",
    "results": Array [
      Object {
        "chance": 70,
        "item": "Меч Убийца Драконов +10",
      },
      Object {
        "chance": 30,
        "item": "Nothing",
      },
    ],
  },
  "Меч Убийца Драконов +10" => Object {
    "cost": "Свиток Модификации Убийц Драконов",
    "item": "Меч Убийца Драконов +10",
    "results": Array [
      Object {
        "chance": 70,
        "item": "Меч Убийца Драконов +11",
      },
      Object {
        "chance": 30,
        "item": "Nothing",
      },
    ],
  },
  "Меч Убийца Драконов +11" => Object {
    "cost": "Свиток Модификации Убийц Драконов",
    "item": "Меч Убийца Драконов +11",
    "results": Array [
      Object {
        "chance": 70,
        "item": "Меч Убийца Драконов +12",
      },
      Object {
        "chance": 30,
        "item": "Nothing",
      },
    ],
  },
  "Меч Убийца Драконов +12" => Object {
    "cost": "Свиток Модификации Убийц Драконов",
    "item": "Меч Убийца Драконов +12",
    "results": Array [
      Object {
        "chance": 70,
        "item": "Меч Убийца Драконов +13",
      },
      Object {
        "chance": 30,
        "item": "Nothing",
      },
    ],
  },
  "Меч Убийца Драконов +13" => Object {
    "cost": "Свиток Модификации Убийц Драконов",
    "item": "Меч Убийца Драконов +13",
    "results": Array [
      Object {
        "chance": 70,
        "item": "Меч Убийца Драконов +14",
      },
      Object {
        "chance": 30,
        "item": "Nothing",
      },
    ],
  },
  "Меч Убийца Драконов +14" => Object {
    "cost": "Свиток Модификации Убийц Драконов",
    "item": "Меч Убийца Драконов +14",
    "results": Array [
      Object {
        "chance": 70,
        "item": "Меч Убийца Драконов +15",
      },
      Object {
        "chance": 30,
        "item": "Nothing",
      },
    ],
  },
  "Меч Убийца Драконов +15" => Object {
    "cost": "Свиток Модификации Убийц Драконов",
    "item": "Меч Убийца Драконов +15",
    "results": Array [
      Object {
        "chance": 35,
        "item": "Меч Убийца Драконов +16",
      },
      Object {
        "chance": 65,
        "item": "Nothing",
      },
    ],
  },
  "Меч Убийца Драконов +16" => Object {
    "cost": "Свиток Модификации Убийц Драконов",
    "item": "Меч Убийца Драконов +16",
    "results": Array [
      Object {
        "chance": 20,
        "item": "Меч Убийца Драконов +17",
      },
      Object {
        "chance": 80,
        "item": "Nothing",
      },
    ],
  },
}
`;
